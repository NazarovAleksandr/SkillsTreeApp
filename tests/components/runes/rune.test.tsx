import * as React from 'react';
import { shallow } from 'enzyme';
import Rune from '../../../src/components/runes/rune';
import {
    IRune, Rarity, RuneTypes, RunePropertyTypes,
} from '../../../src/components/runes/models';

const rune: IRune = {
    id: '1',
    image: 'url',
    rarity: Rarity.common,
    type: RuneTypes.attack,
    properties: [
        { name: RunePropertyTypes.Evade, value: 10 },
    ],
};

describe('Rune', () => {
    test('Renders not used rune', () => {
        const component = shallow(<Rune isUsed={false} rune={rune} />);

        expect(component).toMatchSnapshot('notUsedRune');
    });
    test('Renders not used rune', () => {
        const component = shallow(<Rune isUsed rune={rune} />);

        expect(component).toMatchSnapshot('usedRune');
    });
});
