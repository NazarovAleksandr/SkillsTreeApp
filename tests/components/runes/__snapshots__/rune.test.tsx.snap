// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rune Renders not used rune: notUsedRune 1`] = `
<div
  className="rune"
>
  <img
    alt="rune"
    className="rune-image"
    src="url"
  />
</div>
`;

exports[`Rune Renders not used rune: usedRune 1`] = `
<div
  className="rune used"
>
  <img
    alt="rune"
    className="rune-image"
    src="url"
  />
</div>
`;
